### PR TITLE
[FIX](array_index)fix array tokenized index parser and array_contains bugs

### PR DESCRIPTION
**Pull request by amorynan**
## Proposed changes
1.  fix array tokenized inverted index write with char_string_reader used by _analyzer
2. fix array_contains semantics as match_equals and fix nested null value match 
Issue Number: close #xxx

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Comment by amorynan** [comment]
run buildall

File: be/src/vec/functions/array/function_array_index.h [hunk 1]
```diff
@@ -77,6 +77,7 @@ struct ArrayCountEqual {
 struct ParamValue {
     PrimitiveType type;
     Field value;
+    [[maybe_unused]] bool is_evaled_inverted_idx = false;
```
Comment (airborne12) [inline be/src/vec/functions/array/function_array_index.h:80]: where should it be used ?

Comment (amorynan) [reply be/src/vec/functions/array/function_array_index.h:80]: it will be in array_match_any

**Review by xiaokang — DISMISSED** [review]
LGTM

**Comment by amorynan** [comment]
run buildall

**Review by xiaokang — APPROVED** [review]
LGTM

**Review by airborne12 — APPROVED** [review]
LGTM

**Pull request merged**